### PR TITLE
#34: Use `project-keeper-maven-plugin`

### LINKED ISSUE #34: Use `project-keeper-maven-plugin`
## Situation

To unify handling or our projects, we use the `project-keeper-maven-plugin`.

## Acceptance Criteria

1. Project uses `project-keeper-maven-plugin`

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #34.

### CODE REVIEW

File: doc/changes/changes_4.0.0.md [hunk 1]
```diff
@@ -22,4 +22,5 @@ The old variant is intentionally not supported anymore to tighten security.
 
 ## Plugin Updates
 
+* Added `project-keeper-maven-plugin:0.4.0`
```
Comment (jakobbraun) [inline doc/changes/changes_4.0.0.md:25]: Group id is missing (see https://github.com/exasol/release-droid/blob/main/doc/user_guide/changes_file_template.md)

File: doc/changes/changes_4.0.0.md [hunk 2]
```diff
@@ -22,4 +22,5 @@ The old variant is intentionally not supported anymore to tighten security.
...
 * Updated `jacoco-maven-plugin:0.8.5` to `0.8.6`
```
Comment (jakobbraun) [inline doc/changes/changes_4.0.0.md:26]: You also added `udf-debugging-java` and the `maven-dependency-plugin`

File: pom.xml [hunk 3]
```diff
@@ -178,14 +199,35 @@
...
+        </execution><execution>
+            <id>report-integration-with-udf</id>
+            <phase>verify</phase>
+            <goals>
+                <goal>report-integration</goal>
+            </goals>
+            <configuration>
+                <dataFile>${project.build.directory}/aggregate-it.exec</dataFile>
+            </configuration>
+        </execution></executions>
```
Comment (jakobbraun) [inline pom.xml:230]: reformat file

Comment (jakobbraun) [reply pom.xml:230]: (project-keeper is not formating)

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/changes/changes_4.0.0.md (hunk 1 above)
Comment (redcatbear) [reply doc/changes/changes_4.0.0.md:25]: Fixed.

File: doc/changes/changes_4.0.0.md (hunk 2 above)
Comment (redcatbear) [reply doc/changes/changes_4.0.0.md:26]: Fixed.

File: pom.xml (hunk 3 above)
Comment (redcatbear) [reply pom.xml]: Reformatted.

File: pom.xml [hunk 4]
```diff
@@ -139,19 +140,49 @@
             <version>4.13.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.exasol</groupId>
+            <artifactId>udf-debugging-java</artifactId>
+            <version>0.3.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (jakobbraun) [inline pom.xml:149]: add to changelog

File: pom.xml [hunk 5]
```diff
@@ -139,19 +140,49 @@
             <version>4.13.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (jakobbraun) [inline pom.xml:143]: add to changelog

**Review by jakobbraun — DISMISSED** [review]
approved with two minor changes

File: pom.xml (hunk 5 above)
Comment (redcatbear) [reply pom.xml:143]: Was accidentally under "Plugins". Fixed.

File: pom.xml (hunk 4 above)
Comment (redcatbear) [reply pom.xml:149]: Added.

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**